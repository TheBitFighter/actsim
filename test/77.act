deftype t1 (int<8>a){}
deftype t2 (int<4>b; int<4>c){}

function tf(t1 t) : t2
{
  t2 tt;

  chp {
    tt.b := t.a{3..0};
    tt.c := t.a{7..4};
    self := tt
  }
}

defproc test () 
{
  t1 t;
  t2 tt;

  chp { t.a := 0b01101010; tt := tf(t); 
  log("%x", t.a, " ", tt.b, " ", tt.c) }
}

test t;
