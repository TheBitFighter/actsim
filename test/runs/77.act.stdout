[                  20] <>  6a 10 6
